### ISSUE TITLE
Unable to change permissions

### ISSUE THREAD

**Issue by ojarabo** [realm/realm-mobile-platform#199]
Hi there,
I am trying to change permissions for all users to have access to a specific realm file owned by another user.

However, every time I am trying to change the permissions, it doesn't work, and in the Realm Object Observer, you can see the following message in the log tab:

> [proxy] internal error. Code undefined, status: undefined, message: SyntaxError: Unexpected token j in JSON at position 0 at Object.parse (native) at Service.webSocketUpgradeHandler (/usr/lib/nodejs/realm-object-server/.build/src/node/services/proxy.js:87:30) at Service.safeUpgrade (/usr/lib/nodejs/realm-object-server/.build/src/node/services/proxy.js:57:12) at emitThree (events.js:116:13) at Server.emit (events.js:194:7) at onParserExecuteCommon (_http_server.js:411:14) at HTTPParser.onParserExecute (_http_server.js:379:5)

Any idea of what is going on?

Some code to replicate it
```java
public static final String AUTH_URL         = "http://" + BuildConfig.OBJECT_SERVER_IP + ":9080/auth";
public static final String REALM_URL = "realm://" + BuildConfig.OBJECT_SERVER_IP + ":9080/~/default";

SyncUser.loginAsync(SyncCredentials.usernamePassword(email, password, false), AUTH_URL, this);
....

SyncConfiguration RealmConfiguration = new SyncConfiguration.Builder(user, REALM_URL).build();
Realm.setDefaultConfiguration(RealmConfiguration);
...

Realm realm = user.getManagementRealm();
realm.executeTransactionAsync(new Realm.Transaction() {
                @Override
                public void execute(Realm realm) {
                    Boolean mayRead = true; // Grant read access
                    Boolean mayWrite = true; // Keep current permission
                    Boolean mayManage = false; // Revoke management access
                    PermissionChange change = new PermissionChange(REALM_URL,
                            "*",
                            mayRead,
                            mayWrite,
                            mayManage);
                    realm.insert(change);
                }
            });
```

**Comment by bigfish24** [comment]
Sorry about this, we are looking into this issue and hopefully will have some info soon.

**Comment by mrackwitz** [comment]
We found a bug when changing permissions from admin users and just shipped a release with a fix for that. (See #190) Please upgrade to version 1.7.5 if that could affect you here as well.

But the problem you're seeing appears to be unrelated from permission changes. We're still investigating what could cause the syntax error. We've seen a few more reports like this with clients connecting from Android, but were unable so far to reproduce the issue ourselves. We're working on improving the logging around this, so that we can get to the bottom of this issue.

In the meantime it would be helpful for us to know which version of the Realm Object Server and Java SDK are you using?

**Comment by mrackwitz** [comment]
Further investigations confirm that there is a bug in the Java client SDK. We're working on it. Please refer to https://github.com/realm/realm-java/issues/4750 for more info.

**Comment by ojarabo** [comment]
@mrackwitz I have just seen that you already got the info in the issue you are referring in your comment, but if you still need more info, let me know.

**Comment by mklb** [comment]
I created an entry in the _admin realm with a nil user pointing to the realm I want to give default read/write permissions to. However the realm dashboard DOES NOT show read/write access as the default.

I just downloaded the "latest" version via the website which is V1.7.4 ?

**Comment by cmelchior** [comment]
This has been fixed with Realm Java 3.3.2
